### DISCUSSION TITLE
How do I save all the models within OutputCollect?

### DISCUSSION THREAD

**Discussion by CJ2407** [facebookexperimental/Robyn#688] [Q&A]
Hi Robyn Community,

I only have a limited knowledge of R, but have been running Robyn since last 4-5 months to create a model. I am working on tunning the model to enhance performance, however I have the following hurdles because of which it will be very useful if somebody can help me with steps for saving OutputCollect in Rdata or JSON file. 

1. Due to CPU and Memory issues, I was running Robyn on R Studio on my company's Google VM, but due to run longer run times, my R Studio used to crash in between and I was losing all results. This problem is solved by using TMUX on VM to run the R script, but once the R script is finished none of the R objects are available to access later only the outputs such as CSVs and excels are available. 

2. Second problem is that I have 20+ variables for 3.5 years of data at daily level, which makes one pagers hard to read for many plots. So, I have learnt to get the individual plots from one pagers so that I can format them to be more readable give the number of variables I have. However, these plots can be created after you select a model from OutputCollect.

So, I want to save all the top models/solutions in OutputCollect in a Rdata or JSON file and I found the below code, however it asks me to select a model. How do I select all the top models in the below robyn_write function?
robyn_write(
 InputCollect,
 OutputCollect,
 select_model = NULL,
 dir = OutputCollect$plot_folder,
 export = TRUE,
 quiet = FALSE
)

**Comment by laresbernardo** [comment]
Hi @CJ2407 
- We are working on reduce the size of the objects so that you can run as much iterations and trials as you want with less computation power required. Actually, you can check [this branch](https://github.com/facebookexperimental/Robyn/pull/687) that will be merged soon to main with a 88% size reduction for `OutputModels`.
- You can run a `for` loop (or equivalent) in which you pass the IDs you want to export to `select_model` parameter. 
- You can also save any R object with `saveRDS()` and load them afterwards with `readRDS()`.
